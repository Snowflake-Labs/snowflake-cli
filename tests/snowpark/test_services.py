from tempfile import NamedTemporaryFile
from unittest import mock


@mock.patch("snowflake.connector.connect")
def test_create_service(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx
    with NamedTemporaryFile() as fh:
        result = runner.invoke(
            [
                "snowpark",
                "services",
                "create",
                "--name",
                "serviceName",
                "--compute_pool",
                "computePoolValue",
                "--spec_path",
                fh.name,
                "--num_instances",
                42,
                "--stage",
                "stagName",
            ]
        )

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_list_service(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "services", "list"])

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_drop_service(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "services", "drop", "serviceName"])

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_service_status(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "services", "status", "serviceName"])

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_service_logs(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(
        [
            "snowpark",
            "services",
            "logs",
            "--container_name",
            "containerName",
            "serviceName",
        ]
    )

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == snapshot
