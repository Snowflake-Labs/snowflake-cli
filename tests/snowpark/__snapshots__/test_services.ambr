# serializer version: 1
# name: test_create_service
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  CREATE STAGE IF NOT EXISTS stagName;
  
  put file:///var/folders/1b/6450dkj56wx01kj9ls9vj6y00000gn/T/tmpltyd8jde @stagName/services/d41d8cd98f00b204e9800998ecf8427e auto_compress=false OVERWRITE = TRUE;
  
  CREATE SERVICE IF NOT EXISTS serviceName
    MIN_INSTANCES = 42
    MAX_INSTANCES = 42
    COMPUTE_POOL =  computePoolValue
    spec=@stagName/services/d41d8cd98f00b204e9800998ecf8427e/tmpltyd8jde;
  '''
# ---
# name: test_drop_service
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  DROP SERVICE serviceName;
  '''
# ---
# name: test_list_service
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  show services;
  '''
# ---
# name: test_service_logs
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  call SYSTEM$GET_SERVICE_LOGS('serviceName', '0', 'containerName');
  '''
# ---
# name: test_service_status
  '''
  use role mockRole;
  use warehouse MockWarehouse;
  use database MockDatabase;
  use schema MockSchema;
  
  CALL SYSTEM$GET_SERVICE_STATUS('serviceName');
  '''
# ---
